## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 502188`
Commit: `b19e9f48d4d50f55e04e3a70968113fadf4ae27d`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>borgmatic</li>
    <li>borgmatic.dist</li>
    <li>prometheus-borgmatic-exporter</li>
    <li>prometheus-borgmatic-exporter.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>borgmatic</li>
    <li>borgmatic.dist</li>
    <li>prometheus-borgmatic-exporter</li>
    <li>prometheus-borgmatic-exporter.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>prometheus-borgmatic-exporter</li>
    <li>prometheus-borgmatic-exporter.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>borgmatic</li>
    <li>borgmatic.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>borgmatic</li>
    <li>borgmatic.dist</li>
    <li>prometheus-borgmatic-exporter</li>
    <li>prometheus-borgmatic-exporter.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>prometheus-borgmatic-exporter</summary>
<pre>      )
    File &quot;/nix/store/dgr0qk6qxmh4ssxvl83hz1pcmz8a4iww-python3.13-waitress-3.0.2/lib/python3.13/site-packages/waitress/server.py&quot;, line 243, in __init__
      self.bind_server_socket()
      ~~~~~~~~~~~~~~~~~~~~~~~^^
    File &quot;/nix/store/dgr0qk6qxmh4ssxvl83hz1pcmz8a4iww-python3.13-waitress-3.0.2/lib/python3.13/site-packages/waitress/server.py&quot;, line 364, in bind_server_socket
      self.bind(sockaddr)
      ~~~~~~~~~^^^^^^^^^^
    File &quot;/nix/store/dgr0qk6qxmh4ssxvl83hz1pcmz8a4iww-python3.13-waitress-3.0.2/lib/python3.13/site-packages/waitress/wasyncore.py&quot;, line 374, in bind
      return self.socket.bind(addr)
             ~~~~~~~~~~~~~~~~^^^^^^
  PermissionError: [Errno 1] Operation not permitted
  
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_http.py::test_metrics_endpoint - requests.exceptions.ConnectionError: HTTPConnectionPool(host=&#x27;localhost&#x27;, p...
=================== 1 failed, 26 passed, 1 warning in 1.53s ====================</pre>
</details>
</details>

--- a/pkgs/by-name/bo/borgmatic/package.nix
+++ b/pkgs/by-name/bo/borgmatic/package.nix
@@ -35,9 +35,14 @@ python3Packages.buildPythonApplication rec {
     ++ optional-dependencies.apprise;
 
   # - test_borgmatic_version_matches_news_version
-  # The file NEWS not available on the pypi source, and this test is useless
+  #   NEWS file not available on the pypi source
+  # - test_log_outputs_includes_error_output_in_exception
+  #   TOCTOU race in log_outputs(): process.poll() returns None in
+  #   raise_for_process_errors but non-None in the while-loop exit check,
+  #   so the error is never raised. Timing-dependent; fails on x86_64-darwin.
   disabledTests = [
     "test_borgmatic_version_matches_news_version"
+    "test_log_outputs_includes_error_output_in_exception"
   ];
 
   nativeBuildInputs = [ installShellFiles ];

--- a/pkgs/by-name/bo/borgmatic/package.nix
+++ b/pkgs/by-name/bo/borgmatic/package.nix
@@ -15,12 +15,12 @@
 }:
 python3Packages.buildPythonApplication rec {
   pname = "borgmatic";
-  version = "2.1.3";
+  version = "2.1.5";
   pyproject = true;
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-mWZQBoQUd+cwOV3QU62XyTCFdFpnBz2BsyImzVAozJE=";
+    hash = "sha256-T0+E6opyfr7zxfP44OlNuhqsdQyi7OdIXiE5r310LaU=";
   };
 
   passthru.updateScript = nix-update-script { };
@@ -31,6 +31,7 @@ python3Packages.buildPythonApplication rec {
       flexmock
       pytestCheckHook
       pytest-cov-stub
+      pytest-timeout
     ]
     ++ optional-dependencies.apprise;
 

